sql/core/benchmarks/SubExprEliminationBenchmark-results.txt

```diff
@@ -3,23 +3,23 @@ Benchmark for performance of subexpression elimination
 ================================================================================================
 
 Preparing data for benchmarking ...
-OpenJDK 64-Bit Server VM 17.0.12+7-LTS on Linux 6.5.0-1025-azure
+OpenJDK 64-Bit Server VM 17.0.14+7-LTS on Linux 6.8.0-1021-azure
 AMD EPYC 7763 64-Core Processor
 from_json as subExpr in Project:          Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-subExprElimination false, codegen: true            6438           6551          98          0.0    64378783.5       1.0X
-subExprElimination false, codegen: false           6216           6320         175          0.0    62161826.1       1.0X
-subExprElimination true, codegen: true             1480           1518          39          0.0    14799890.8       4.3X
-subExprElimination true, codegen: false            1321           1429          94          0.0    13212919.6       4.9X
+subExprElimination false, codegen: true            6398           6601         177          0.0    63980070.0       1.0X
+subExprElimination false, codegen: false           6018           6137         123          0.0    60178371.6       1.1X
+subExprElimination true, codegen: true             1288           1352          78          0.0    12883220.0       5.0X
+subExprElimination true, codegen: false            1299           1340          66          0.0    12989399.4       4.9X
 
 Preparing data for benchmarking ...
-OpenJDK 64-Bit Server VM 17.0.12+7-LTS on Linux 6.5.0-1025-azure
+OpenJDK 64-Bit Server VM 17.0.14+7-LTS on Linux 6.8.0-1021-azure
 AMD EPYC 7763 64-Core Processor
 from_json as subExpr in Filter:           Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-subExprElimination false, codegen: true            7107           7310         207          0.0    71066752.8       1.0X
-subExprElimination false, codegen: false           6738           6781          41          0.0    67375897.0       1.1X
-subExprElimination true, codegen: true             2052           2110          51          0.0    20519152.3       3.5X
-subExprElimination true, codegen: false            2053           2079          33          0.0    20526629.8       3.5X
+subExprElimination false, codegen: true            2474           2512          44          0.0    24744107.3       1.0X
+subExprElimination false, codegen: false           2231           2246          20          0.0    22306061.2       1.1X
+subExprElimination true, codegen: true             2408           2509         100          0.0    24084091.2       1.0X
```

Comment:
What confuses me is that when `subExprElimination` set to true, `codegen` set to true, there is an obvious regression in performance compared to before?

Reply:
Yes, it's true. The performance regression was the actual root cause why we didn't make a decision. Here, FYI.

- https://github.com/apache/spark/pull/49411